lib/fluent/plugin/in_tail.rb

```diff
@@ -615,11 +625,13 @@ def detach_watcher_after_rotate_wait(tw, ino)
             detach_watcher(tw, ino)
           end
         end
+        @tails_rotate_wait[tw] = { ino: ino, timer: timer }
       else
         # when the throttling feature isn't enabled, just wait @rotate_wait
-        timer_execute(:in_tail_close_watcher, @rotate_wait, repeat: false) do
+        timer = timer_execute(:in_tail_close_watcher, @rotate_wait, repeat: false) do
           detach_watcher(tw, ino)
         end
+        @tails_rotate_wait[tw] = { ino: ino, timer: timer }
```

Comment:
Why is it necessary to keep these `timer` objects?

Reply:
In my understanding, the complete process of cancelling roate_wait needs `@tails_rotate_wait[tw][:timer].detach)`.
If we need to cancel rotaet_wait at certain timing, we need to keep the timer ID.
Ideally it's also needed in `shutdown` but this PR omits it since event_loop will be stopped after shutdown so that the timer never called.

Reply:
I see! Thanks!

> If we need to cancel rotaet_wait at certain timing, we need to keep the timer ID (we never cancel it in the current implementation though).

Yes!

> Ideally it's also needed in shutdown but this PR omits it since event_loop will be stopped after shutdown so that the timer never called.

Yes! `event_loop` will detach them automatically, so we can omit it.